File: ProductExtractor/database/category_repository.py
from typing import Optional
from database.utils import get_connection, insert, select
from category import Category
from database.repository import Repository


class CategoryRepository(Repository[Category]):
    @classmethod
    def tablename(cls) -> str:
        return "Category"

    @classmethod
    def add(cls, item: Category) -> int:
        rows = select(cls.tablename(), ["ID"], {"Name": item.name})

        if len(rows) > 0:
            return rows[0][0]

        return insert(
            cls.tablename(),
            {
                "Name": item.name,
            },
        )

File: src/Repositories/CategoryRepository.php
<?php

namespace App\Repositories;
use App\Entities\Category;

use Exception;

class CategoryRepository extends Repository {
    protected static string $tableName = "Category";

    public static function getByID(int $categoryID): Category {
        $data = self::select(['ID' => $categoryID])[0];
        return self::convertToCategory($data);
    }

    private static function convertToCategory(object $data): Category {
        return new Category(
            $data->ID,
            $data->Name,
        );
    }
}
